source/common/ssl/context_manager_impl.cc

```diff
@@ -47,7 +137,7 @@ size_t ContextManagerImpl::daysUntilFirstCertExpires() {
 }
 
 void ContextManagerImpl::iterateContexts(std::function<void(Context&)> callback) {
-  std::unique_lock<std::mutex> lock(contexts_lock_);
+  std::shared_lock<std::shared_timed_mutex> lock(contexts_lock_);
```

Comment:
Same comment about function const-ness as above

Reply:
Actually, if the callback takes a non-const reference, this should probably a non-shared lock.

Reply:
Fixed.